### ISSUE TITLE
please add feature to bulk delete

### ISSUE THREAD

**Issue by ramstein74** [deployd/deployd#488]
hello, this is the way i know to delete all records from collection

dpd.collection.get(function(result,error){
result.forEach(function(entry) {
dpd.collection.del(entry.id,function(){});
});
})

There should be a better way right?

**Comment by simpleTechs** [comment]
While this is most certainly not perfect, what you can do right now is supply a query to `del` - this will not require you to loop over a collection and call it again and again.

Imagine the following scenario: You have an object, say Class, that has many childs called Student. (This relation is setup according to http://docs.deployd.com/docs/collections/relationships-between-collections.html#s-Contains%20Many%20or%20One-to-Many-2191)

So on every Class-instance, you will find an array of "Student-Ids":
```js
{ 
  name: 'Testclass',
  students: [ 'ceb59a120693e8b8', '88acc28d9d7a0973' ],
  id: 'b039e99a43e1f82e'
}
```

Each Student on its own will have some properties, but that's not so important right now:
```js
[
  { name: 'Student A', id: 'ceb59a120693e8b8' },
  { name: 'Student B', id: '88acc28d9d7a0973' }
]
```

What you can do now is simple: construct a query that will select all the students and pass it to the `del` function:
```js
// get them through a query, for example
var studentIds = [ 'ceb59a120693e8b8', '88acc28d9d7a0973' ];

dpd.students.get({ id: { $in: studentIds }}, function(res, err) {
    console.log('get before delete:', res);

    // note: this is only called once, and it will delete more than one object
    dpd.students.del({ id: { $in: studentIds }}, function(res, err) {
        console.log('delete response:', res);
    
        dpd.students.get({ id: { $in: studentIds }}, function(res, err) {
            console.log('get after delete:', res);
        });
    });
});
```

The output will be as following:
```
  get before delete: [ 
    { name: 'Student A', id: 'ceb59a120693e8b8' },
    { name: 'Student B', id: '88acc28d9d7a0973' }
  ]
  delete response: undefined
  get after delete: []
```

This is how you can easily delete in bulk using a query. One thing to keep in mind is that deployd will require an id in the query. This *somehow* makes sense, to limit the outcome of your delete, but there's an easy way to get around this, say you wanted to delete every Student named `Student A`:

```js
dpd.students.get({ name: 'Student A' }, function(res, err) {
    console.log('get before delete:', res);
    
    dpd.students.del({ name: 'Student A' }, function(res, err) {
        console.log('delete response:', err);
    
        dpd.students.get({ name: 'Student A' }, function(res, err) {
            console.log('get after delete:', res);
        });
    });
});
```

Now dpd will tell you: `delete response: {"message":"You must include a query with an id when deleting an object from a collection.","status":400}`
Let's pretend you don't know the Ids (which is not quite true in this example, but anyway) - what you can do now is fool dpd into thinking you know them, by filtering on the id field. In reality this filter is saying "select every object that has an id set" - which is every single object in your database. So **make sure** to have a good **AND** condition set, or you **will loose data**.

```js
dpd.students.get({ name: 'Student A' }, function(res, err) {
    console.log('get before delete:', res);
    
    dpd.students.del({ name: 'Student A', id: { $ne: 'null' } }, function(res, err) {
        console.log('delete response:', res);
    
        dpd.students.get({ name: 'Student A' }, function(res, err) {
            console.log('get after delete:', res);
        });
    });
});
```

dpd will now happily go ahead, query the students by name and then delete all of them in one single query. This is how you delete in bulk when not knowing the ids in advance.

**Comment by ramstein74** [comment]
Very detailed info. Thank u but i just dont understand something

the delete query is not using the return (res) from the get.

**Comment by simpleTechs** [comment]
Yes, that's what this was about: I only put in the `get`-queries to show what the database looks like before and after the `delete` call, you can easily throw them out and it will still work just fine.

**Comment by andreialecu** [comment]
I think this will also bulk delete everything:

```javascript
dpd.collection.del({ id: { $ne: 'null' } }
```

Be extremely careful with it, make sure you have proper validation checks in the DELETE event so that you don't accidentally delete something you shouldn't.

**Comment by NicolasRitouet** [comment]
Close since fixed by #502 
Reopen if necessary
